### PR TITLE
Feature/api platform endpoint

### PR DESCRIPTION
**Pull request by tgrochowski**

### CODE REVIEW

File: tests/Behat/Context/Api/WishlistContext.php [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+        ]);
+
+        return $this->client->request(
+            self::DELETE,
+            sprintf('%s%s', self::$domain, $uri),
+            $this->getOptions(self::DELETE)
+        );
+    }
+
+    /** @Given user :email :password is authenticated
```
Comment (patrick477) [inline tests/Behat/Context/Api/WishlistContext.php:145]: format

File: tests/Behat/Context/Api/WishlistContext.php [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+     * @throws GuzzleException
+     */
+    public function userIsAuthenticated(string $email, string $password): void
+    {
+        $uri = '/api/v2/shop/authentication-token';
```
Comment (patrick477) [inline tests/Behat/Context/Api/WishlistContext.php:150]: route name?

File: tests/Behat/Context/Api/WishlistContext.php [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+        Assert::eq($response->getStatusCode(), 200);
+
+        $json = json_decode((string)$response->getBody());
+
+        $this->user = $this->userRepository->findOneByEmail($email);
+        $this->token = (string)$json->token;
+    }
+
+    /** @Given user is unauthenticated */
+    public function userIsUnauthenticated()
```
Comment (patrick477) [inline tests/Behat/Context/Api/WishlistContext.php:179]: return type?

File: tests/Behat/Context/Api/WishlistContext.php [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+
+    /**
+     * @BeforeScenario
+     */
+    public function setupDomain()
+    {
+        $domain = (string)$this->getMinkParameter("base_url");;
+        self::$domain = trim($domain, "/");
+    }
+}
```
Comment (patrick477) [inline tests/Behat/Context/Api/WishlistContext.php:354]: - return type is missing
- public methods should be ahead of private ones
- var_dump to remove 
- behat annotations have wrong format, see how it is on other plugins
- why you sometimes use apostdores and then quotes, stick to one format
- ECS is missing

File: tests/Behat/Resources/suites.yml [hunk 5]
```diff
@@ -18,3 +18,29 @@ default:
...
+
+                - sylius.behat.context.setup.user
+
+                - sylius.behat.context.transform.product_variant
+
+                - sylius.behat.context.ui.shop.product
+
+                - bitbag_wishlist_plugin.behat.context.api.wishlist
+            filters:
+                tags: "@api_wishlist && @api"
```
Comment (patrick477) [inline tests/Behat/Resources/suites.yml:46]: extract to separate file

Comment (patrick477) [reply tests/Behat/Resources/suites.yml:46]: context grouping is missing on setup, transformer etc

File: tests/Application/config/packages/framework.yaml [hunk 6]
```diff
@@ -4,3 +4,6 @@ framework:
     csrf_protection: true
     session:
         handler_id: ~
+    serializer:
+        mapping:
+            paths: ['%kernel.project_dir%/../../src/Resources/config/serialization']
```
Comment (patrick477) [inline tests/Application/config/packages/framework.yaml:9]: will it work like someone installs a plugin in a project? is there any description in readme?

File: src/Resources/config/services/message_handler.yml [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        tags:
+            - { name: messenger.message_handler }
+
+    bitbag.sylius_wishlist_plugin.command_handler.wishlist.remove_wishlist:
+        class: BitBag\SyliusWishlistPlugin\CommandHandler\Wishlist\RemoveWishlistHandler
+        arguments:
+            - "@bitbag_sylius_wishlist_plugin.repository.wishlist"
+            - "@bitbag_sylius_wishlist_plugin.manager.wishlist"
+        tags:
+            - { name: messenger.message_handler}
```
Comment (patrick477) [inline src/Resources/config/services/message_handler.yml:56]: use one service id convention, fix all services to one format

File: src/EventListener/MergeUserWishlistItemsListener.php [hunk 8]
```diff
@@ -70,13 +70,13 @@ private function resolveWishlist(Request $request, ShopUserInterface $shopUser):
...
 
-        if (null !== $cookieWishlist && null !== $userWishlist) {
+        if (null !== $userWishlist) {
             foreach ($cookieWishlist->getWishlistProducts() as $wishlistProduct) {
                 $userWishlist->addWishlistProduct($wishlistProduct);
             }
         }
 
-        if (null !== $cookieWishlist && null === $userWishlist) {
+        if (null === $userWishlist) {
```
Comment (patrick477) [inline src/EventListener/MergeUserWishlistItemsListener.php:79]: will it work outside of the API?

**Review by patrick477 — CHANGES_REQUESTED** [review]

File: src/EventListener/MergeUserWishlistItemsListener.php (hunk 8 above)
Comment (tgrochowski) [reply src/EventListener/MergeUserWishlistItemsListener.php:79]: https://github.com/TymekB ? 

@patrick477 on line 67 there is a check for $cookieWishlist

**Review by patrick477 — APPROVED** [review]

**Pull request merged**